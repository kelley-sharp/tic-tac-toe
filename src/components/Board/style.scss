.board {
  margin: 2em;
  display: flex;
  flex-flow: column nowrap;
}

.board-button-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 1em;
}

.row {
  display: flex;
  flex-flow: row nowrap;
  flex: 1;
}
