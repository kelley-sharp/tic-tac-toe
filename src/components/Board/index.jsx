import React, { PureComponent } from 'react';
import './style.scss';
import Square from '../Square';

class Board extends PureComponent {
  handleClick = (row, col) => {
    //if board is not frozen (gameOver) and there is no value on that square, take a turn
    if (!this.props.frozen && this.props.board[row][col] === null) {
      this.props.handleTurn(row, col);
    }

    //if board is frozen, the square's value remains null
    if (this.props.frozen) {
      this.props.board[row][col] = null;
    }
  };

  render() {
    return (
      <div className="board">
        {this.props.board.map((row, rowIdx) => {
          return (
            <div className="row" key={`row${rowIdx}`}>
              {row.map((squareVal, colIdx) => {
                return (
                  <Square
                    handleClick={() => this.handleClick(rowIdx, colIdx)}
                    id={`square${rowIdx}${colIdx}`}
                    rowIdx={rowIdx}
                    colIdx={colIdx}
                    value={squareVal}
                    key={`square${rowIdx}${colIdx}`}
                  />
                );
              })}
            </div>
          );
        })}
      </div>
    );
  }
}

export default Board;
