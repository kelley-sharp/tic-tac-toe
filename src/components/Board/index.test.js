import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Board from '.';

describe('<Board />', () => {
  let wrapper;
  it('renders without crashing', () => {
    wrapper = shallow(
      <Board
        board={[[null, null, null], [null, null, null], [null, null, null]]}
        frozen={true}
        handleTurn={jest.fn()}
      />
    );
  });

  it('matches snapshot', function() {
    wrapper = shallow(
      <Board
        board={[[null, null, null], [null, null, null], [null, null, null]]}
        frozen={true}
        handleTurn={jest.fn()}
      />
    );
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
