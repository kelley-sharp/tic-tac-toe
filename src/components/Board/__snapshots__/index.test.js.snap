// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> matches snapshot 1`] = `
<div
  className="board"
>
  <div
    className="row"
    key="row0"
  >
    <Square
      colIdx={0}
      handleClick={[Function]}
      id="square00"
      key="square00"
      rowIdx={0}
      value={null}
    />
    <Square
      colIdx={1}
      handleClick={[Function]}
      id="square01"
      key="square01"
      rowIdx={0}
      value={null}
    />
    <Square
      colIdx={2}
      handleClick={[Function]}
      id="square02"
      key="square02"
      rowIdx={0}
      value={null}
    />
  </div>
  <div
    className="row"
    key="row1"
  >
    <Square
      colIdx={0}
      handleClick={[Function]}
      id="square10"
      key="square10"
      rowIdx={1}
      value={null}
    />
    <Square
      colIdx={1}
      handleClick={[Function]}
      id="square11"
      key="square11"
      rowIdx={1}
      value={null}
    />
    <Square
      colIdx={2}
      handleClick={[Function]}
      id="square12"
      key="square12"
      rowIdx={1}
      value={null}
    />
  </div>
  <div
    className="row"
    key="row2"
  >
    <Square
      colIdx={0}
      handleClick={[Function]}
      id="square20"
      key="square20"
      rowIdx={2}
      value={null}
    />
    <Square
      colIdx={1}
      handleClick={[Function]}
      id="square21"
      key="square21"
      rowIdx={2}
      value={null}
    />
    <Square
      colIdx={2}
      handleClick={[Function]}
      id="square22"
      key="square22"
      rowIdx={2}
      value={null}
    />
  </div>
</div>
`;
